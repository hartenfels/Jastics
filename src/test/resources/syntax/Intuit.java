public class Intuit from "music.rdf" {
  public static void test() {
    ⊤ top;
    ⊥ bot;

    ¬⊤  nt;
    ¬¬⊤ nnt;

    ⊤ ⊔ ⊥ tub;
    ⊤ ⊓ ⊥ tnb;

    ∃△·«:MusicArtist» e1;
    ∀▽·⎨«:hendrix»⎬   a1;

    ∃▽·⊤    e2;
    ∀△⁻·¬¬⊥ a2;

    ∃▽·⊤ ⊔ «:MusicArtist»   e3;
    ∃▽·⊤ ⊔ ⊤ ⊔ ⊤ ⊓ ⊥        e4;
    ⊤ ⊔ ⊥ ⊔ ∃▽·⎨«:beatles»⎬ e5;
    ⊤ ⊔ ⊤ ⊓ ⊥ ⊔ ∃▽·⊤ ⊓ ⊥    e6;
  }

  public static void testTexas() {
    #T top;
    #F bot;

    -.#T   nt;
    -.-.#T nnt;

    #T ||| #F tub;
    #T &&& #F tnb;

    #E#f...<<<:MusicArtist>>> e1;
    #A#t...{|<<<:hendrix>>>|} a1;

    #E#t...#T      e2;
    #A#f⁻...-.-.#F a2;

    #E#t...#T ||| <<<:MusicArtist>>>        e3;
    #E#t...#T ||| #T ||| #T &&& #F          e4;
    #T ||| #F ||| #E#t...{|<<<:beatles>>>|} e5;
    #T ||| #T &&& #F ||| #E#t...#T &&& #F   e6;
  }
}
