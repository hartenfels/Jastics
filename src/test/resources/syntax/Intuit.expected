public class Intuit from "music.rdf" {
  public static void test() {
    ⊤ top;
    ⊥ bot;
    ¬⊤ nt;
    ¬¬⊤ nnt;
    ⊤ ⊔ ⊥ tub;
    ⊤ ⊓ ⊥ tnb;
    ∃△·«:MusicArtist» e1;
    ∀▽·⎨«:hendrix»⎬ a1;
    ∃▽·⊤ e2;
    ∀△⁻·¬¬⊥ a2;
    ∃▽·⊤ ⊔ «:MusicArtist» e3;
    ∃▽·⊤ ⊔ ⊤ ⊔ ⊤ ⊓ ⊥ e4;
    ⊤ ⊔ ⊥ ⊔ ∃▽·⎨«:beatles»⎬ e5;
    ⊤ ⊔ ⊤ ⊓ ⊥ ⊔ ∃▽·⊤ ⊓ ⊥ e6;
  }

  public static void testTexas() {
    ⊤ top;
    ⊥ bot;
    ¬⊤ nt;
    ¬¬⊤ nnt;
    ⊤ ⊔ ⊥ tub;
    ⊤ ⊓ ⊥ tnb;
    ∃△·«:MusicArtist» e1;
    ∀▽·⎨«:hendrix»⎬ a1;
    ∃▽·⊤ e2;
    ∀△⁻·¬¬⊥ a2;
    ∃▽·⊤ ⊔ «:MusicArtist» e3;
    ∃▽·⊤ ⊔ ⊤ ⊔ ⊤ ⊓ ⊥ e4;
    ⊤ ⊔ ⊥ ⊔ ∃▽·⎨«:beatles»⎬ e5;
    ⊤ ⊔ ⊤ ⊓ ⊥ ⊔ ∃▽·⊤ ⊓ ⊥ e6;
  }
}
